pkg/vm/vm_test.go

```diff
@@ -330,6 +342,76 @@ func TestComputeAddRes(t *testing.T) {
 	assert.Equal(t, expected, res)
 }
 
+func TestComputeAddRes_Jake(t *testing.T) {
+	// Test cases
+	testCases := []struct {
+		name       string
+		op0        *mem.MemoryValue
+		op1        *mem.MemoryValue
+		expected   *mem.MemoryValue
+		shouldFail bool
+	}{
+		{
+			name:     "Type A : MemoryAddressPlusInteger",
+			op0:      mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10)),
+			op1:      mem.MemoryValueFromInt(15),
+			expected: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 25)),
+		},
+		{
+			name:     "Type B : MemoryAddressPlusMemoryAddress_SameSegment",
+			op0:      mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10)),
+			op1:      mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 15)),
+			expected: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 25)),
```

Comment:
This should crash, since address + address is not allowed

Reply:
Okay. Will modify it.